src/ess/nmx/mcstas/executables.py

```diff
@@ -0,0 +1,274 @@
+        dynamic_keys=(RawEventProbability,),
+        target_keys=(NMXRawDataMetadata,),
+        accumulators={
+            MaximumProbability: MaxAccumulator,
+            MaximumTimeOfArrival: MaxAccumulator,
+            MinimumTimeOfArrival: MinAccumulator,
+        },
+    )
+
+
+def _build_final_streaming_processor_helper() -> (
+    Callable[[sl.Pipeline], StreamProcessor]
+):
+    return partial(
+        StreamProcessor,
+        dynamic_keys=(RawEventProbability,),
+        target_keys=(NMXReducedDataGroup,),
+        accumulators={NMXReducedCounts: EternalAccumulator},
+    )
+
+
+def calculate_raw_data_metadata(
+    *detector_ids: DetectorIndex | DetectorName,
+    wf: sl.Pipeline,
+    chunk_size: int = 10_000_000,
+    logger: logging.Logger | None = None,
+) -> NMXRawDataMetadata:
+    # Stream processor building helper
+    scalefactor_stream_processor = _build_metadata_streaming_processor_helper()
+    metadata_wf = wf.copy()
+    # Loop over the detectors
+    file_path = metadata_wf.compute(FilePath)
+    raw_data_metadatas = {}
+
+    for detector_i in detector_ids:
+        temp_wf = metadata_wf.copy()
+        if isinstance(detector_i, str):
+            temp_wf[DetectorName] = detector_i
+        else:
+            temp_wf[DetectorIndex] = detector_i
+
+        detector_name = temp_wf.compute(DetectorName)
+        max_chunk_id = calculate_number_of_chunks(
+            temp_wf.compute(FilePath),
+            detector_name=detector_name,
+            chunk_size=chunk_size,
+        )
+        # Build the stream processor
+        processor = scalefactor_stream_processor(temp_wf)
+        for i_da, da in enumerate(
+            raw_event_data_chunk_generator(
+                file_path=file_path, detector_name=detector_name, chunk_size=chunk_size
+            )
+        ):
+            if any(da.sizes.values()) == 0:
+                continue
+            else:
+                results = processor.add_chunk({RawEventProbability: da})
+            if logger is not None:
+                logger.info(
+                    "[{%s}/{%s}] Processed chunk for {%s}",
+                    i_da + 1,
+                    max_chunk_id,
+                    detector_name,
+                )
+
+        raw_data_metadatas[detector_i] = results[NMXRawDataMetadata]
+
+    # We take the min/maximum values of the scale factor
+    # We are doing it manually because it is not possible to update parameters
+    # in the workflow that stream processor uses.
+    min_toa = min(dg.min_toa for dg in raw_data_metadatas.values())
+    max_toa = max(dg.max_toa for dg in raw_data_metadatas.values())
+    max_probability = max(dg.max_probability for dg in raw_data_metadatas.values())
+
+    return NMXRawDataMetadata(
+        min_toa=min_toa, max_toa=max_toa, max_probability=max_probability
+    )
+
+
+def reduction(
+    *,
+    input_file: pathlib.Path,
+    output_file: pathlib.Path,
+    chunk_size: int = 10_000_000,
+    detector_ids: list[int],
```

Comment:
```suggestion
    detector_ids: list[int | str],
``` 
?
The type checkers seems to be complaining about `detector_ids` too as the `main` function forces it to be integer IDs and not detector names.